core/services/ocr2/plugins/median/services.go

```diff
@@ -145,24 +125,10 @@ func NewMedianServices(ctx context.Context,
 		CreatedAt:    time.Now(),
 	}, lggr)
 
-	medianPluginCmd := env.MedianPluginCmd.Get()
-	medianLoopEnabled := medianPluginCmd != ""
-
-	// TODO BCF-2821 handle this properly as this blocks Solana chain reader dev
-	if !medianLoopEnabled && medianProvider.ChainReader() != nil {
```

Comment:
Don't we still needed this if block?  The TODO comment is just referring to the `!medianLoopEnabled` check, which is unnecessary now that `medianProvider.ChainReader()` can properly return nil even when LOOP is enabled.

Reply:
Oh, nevermind... I see it's just been moved to chainlink-feeds.  We should probably have n info log there "ChainReader enabled" or "ChainReader disabled" to replace this, but this PR looks good.

Reply:
This [PR](https://github.com/smartcontractkit/chainlink-feeds/pull/7/files) should cover it

Reply:
The reason this did not previously trigger the nil pointer that we are seeing now, is that the `medianLoopEnabled` check is backwards. It should have been `medianLoopEnabled && medianProvider.ChainReader() != nil` - not `!medianLoopEnabled`. The nil check only fails in LOOPP mode in the first place, because we make a grpc client.
@reductionista

Reply:
Oh was this meant to be an intentional flag? This extremely confusing.

Reply:
It wasn't backwards, but yeah I think this comment was probably the confusing thing here: 
```
// TODO BCF-2821 handle this properly as this blocks Solana chain reader dev
```
Would have been more clear if it had said "once BCF-2821 is merged we can remove the !medianLoopEnabled check".  Any PR's raised against develop or main need to preserve that the way it is.

Originally it was just the nil check, the JIRA was created along with the comment and the addition of the !medianLoopEnabled so that it would always be disabled for LOOP until we fixed the nil issue with LOOP.  None of us have tested it with LOOP enabled yet so that wasn't a priority, but hopefully we can start testing it that way soon.

Reply:
Oh, also my initial comment was wrong...  might have been even more confusing.  I said that this check was unnecessary now, but it didn't occur to me to think about which branch we were on... necessary on this branch but not on the one we've been working on.